### PR TITLE
Add support for WiFi7 802.11be modes

### PR DESCRIPTION
**Pull request by lakdif**
80/160/240/320Mhz - EHT80,EHT160,EHT240,EHT320
Hostapd must be updated as well to provide those values

- [x] This PR is not from my *main* or *master* branch :poop:, but a *separate* branch :white_check_mark:
- [x] Each commit has a valid :black_nib: `Signed-off-by: <my@email.address>` row (via `git commit --signoff`)
- [x] Each commit and PR title has a valid :memo: `<package name>: title` first line subject for packages
- [x] Incremented :up: any `PKG_VERSION` in the Makefile
- [x] Tested on: (architecture, openwrt version, browser) :white_check_mark:
- [x] \( Preferred ) Mention: @ the original code author for feedback
- [x] \( Preferred ) Screenshot or mp4 of changes:
- [x] \( Optional ) Closes: e.g. openwrt/luci#issue-number
- [x] \( Optional ) Depends on: e.g. openwrt/packages#pr-number in sister repo
- [x] Description: (describe the changes proposed in this PR)

### CODE REVIEW

**Comment by hnyman** [comment]
Have you tested this?
(as it seems to have syntax errors, missing commas, I doubt that you have...)

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 1]
```diff
@@ -362,6 +363,12 @@ var CBIWifiFrequencyValue = form.Value.extend({
 					'HE80', '80 MHz', htmodelist.HE80,
 					'HE160', '160 MHz', htmodelist.HE160
 				]
+				'be': [
```
Comment (hnyman) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:366]: a comma missing before this line?

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 2]
```diff
@@ -382,6 +389,12 @@ var CBIWifiFrequencyValue = form.Value.extend({
 					'5g', '5 GHz', this.channels['5g'].length > 3,
 					'6g', '6 GHz', this.channels['6g'].length > 3
 				]
+				'be': [
```
Comment (hnyman) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:392]: a comma missing before this line?

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 3]
```diff
@@ -445,7 +458,9 @@ var CBIWifiFrequencyValue = form.Value.extend({
 
 		this.setValues(mode, this.modes);
 
-		if (/HE20|HE40|HE80|HE160/.test(htval))
+		if (/EHT80|EHT160/EHT240/EHT320/.test(htval))
```
Comment (hnyman) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:461]: Shouldn't it be | between values instead of / ?

**Comment by wlshdjj** [comment]
I think there is a mistake at line 461:

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 1 above)
Comment (hnyman) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:366]: After "]" on the previous line. 
...like with all the other list elements.

**Comment by rmandrad** [comment]
so I tested this as I have a wifi7 device (banana pi4 be14) ... it doesn't show BE in the mode list 

this is a snippet of my board.json 

`                "phy2": {
                        "path": "soc/11300000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0+2",
                        "info": {
                                "antenna_rx": 7,
                                "antenna_tx": 7,
                                "bands": {
                                        "6G": {
                                                "he": true,
                                                "eht": true,
                                                "max_width": 320,
                                                "modes": [
                                                        "NOHT",
                                                        "HE20",
                                                        "EHT20",
                                                        "HE80",
                                                        "EHT80",
                                                        "HE160",
                                                        "EHT160",
                                                        "EHT320"
                                                ],
                                                "default_channel": 1
`

**Comment by rmandrad** [comment]
it doesn't work I found a few more typos

also modules/luci-base/root/usr/share/rpcd/ucode/luci & modules/luci-base/htdocs/luci-static/resources/network.js will need to change and I guess also iwinfo

**Comment by rmandrad** [comment]
`--- a/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
@@ -338,7 +338,8 @@ var CBIWifiFrequencyValue = form.Value.extend({
                                '', 'Legacy', hwmodelist.a || hwmodelist.b || hwmodelist.g,
                                'n', 'N', hwmodelist.n,
                                'ac', 'AC', L.hasSystemFeature('hostapd', '11ac') && hwmodelist.ac,
-                               'ax', 'AX', L.hasSystemFeature('hostapd', '11ax') && hwmodelist.ax
+                               'ax', 'AX', L.hasSystemFeature('hostapd', '11ax') && hwmodelist.ax,
+                               'be', 'BE', L.hasSystemFeature('hostapd', '11be') && hwmodelist.be
                        ];
 
                        var htmodelist = L.toArray(data[0] ? data[0].getHTModes() : null)
@@ -361,7 +362,13 @@ var CBIWifiFrequencyValue = form.Value.extend({
                                        'HE40', '40 MHz', htmodelist.HE40,
                                        'HE80', '80 MHz', htmodelist.HE80,
                                        'HE160', '160 MHz', htmodelist.HE160
-                               ]
+                               ],
+                               'be': [
+                                       'EHT80', '80 MHz', htmodelist.EHT80,
+                                       'EHT160', '160 MHz', htmodelist.EHT160,
+                                       'EHT240', '240 MHz', htmodelist.EHT240,
+                                       'EHT320', '320 MHz', htmodelist.EHT320
+                ]
                        };
 
                        this.bands = {


`

`--- a/modules/luci-base/root/usr/share/rpcd/ucode/luci
+++ b/modules/luci-base/root/usr/share/rpcd/ucode/luci
@@ -208,7 +208,7 @@ const methods = {
                                relayd:     access('/usr/sbin/relayd') == true,
                        };
 
-                       const wifi_features = [ 'eap', '11ac', '11ax', '11r', 'acs', 'sae', 'owe', 'suiteb192', 'wep', 'wps', 'ocv' ];
+                       const wifi_features = [ 'eap', '11ac', '11ax', '11be', '11r', 'acs', 'sae', 'owe', 'suiteb192', 'wep', 'wps', 'ocv' ];
 
                        if (access('/usr/sbin/hostapd')) {
                                result.hostapd = { cli: access('/usr/sbin/hostapd_cli') == true };
`

--- a/modules/luci-base/htdocs/luci-static/resources/network.js
+++ b/modules/luci-base/htdocs/luci-static/resources/network.js
@@ -3374,7 +3374,7 @@ WifiDevice = baseclass.extend(/** @lends LuCI.network.WifiDevice.prototype */ {
         */
        getHWModes: function() {
                var hwmodes = this.ubus('dev', 'iwinfo', 'hwmodes');
-               return Array.isArray(hwmodes) ? hwmodes : [ 'b', 'g' ];
+               return Array.isArray(hwmodes) ? hwmodes : [ 'b', 'g', 'a' ];
        },
 
        /**

**Comment by rmandrad** [comment]
i closed the pull - https://github.com/openwrt/luci/pull/7277

**Comment by systemcrash** [comment]
Closing in favour of other PR which works and has been tested.

**Pull request closed without merging**